Tech-Journal-ConsoleApp/Tech-Journal-ConsoleApp/EmailEntry.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.IO;
+using System.Net;
+using System.Net.Mail;
+using System.Text;
+
+namespace Tech_Journal_ConsoleApp
+{
+    class EmailEntry
+    {
+        public const string Path = "c:\\temp\\appsettings.txt";
+
+        private bool _validEmail;
+        private string _toThisEmailAddress;
+        private string _emailUsername;
+        private string _emailPassword;
+
+
+        public EmailEntry()
+        {
+            _validEmail = false;
+        }
+
+        public bool CheckforEmailSettings()
+        {
+            if (!File.Exists(Path))
+            {
+                return false;
+            }
+            else
+            {
+                return true;
+            }
+        }
+
+        public void GenerateSenderEmailSettings()
+        {
+            using (var sw = File.AppendText(Path))
+            {
+                Console.WriteLine("Email Settings were not found. Creating file to store email settings.");
+                while (!_validEmail)
+                {
+                    Console.WriteLine("Please enter an email address to send the Journal entry from");
+                    _emailUsername = Console.ReadLine();
+                    _validEmail = IsValidEmailAddress(_emailUsername);
+                }
+                sw.WriteLine(_emailUsername);
+                Console.WriteLine("Please enter the password for that email address");
+                _emailPassword = Console.ReadLine();
+                sw.WriteLine(_emailPassword);
+                _validEmail = false;
+            }
+        }
+
+        public void ReadSenderEmailSettings()
+        {
+            using (var emailSettings = new StreamReader(Path))
+            {
+                _emailUsername = emailSettings.ReadLine();
+                _emailPassword = emailSettings.ReadLine();
+            }
+        }
+        
+        public void SendEmail(string entry, string userName)
+        {
+
+            while (!_validEmail)
```

Comment:
You could pull this while loop into it's own method. It could be something like `GetValidEmailAddress()` and would return the valid email string. Then you can set the return to the variable you need to set it to.